pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -454,17 +462,48 @@ stdenv.mkDerivation ({
     runHook postInstall
   '';
 
-  passthru = passthru // {
+  passthru = passthru // rec {
 
     inherit pname version;
 
     compiler = ghc;
 
-
     getBuildInputs = {
-      inherit propagatedBuildInputs otherBuildInputs allPkgconfigDepends;
-      haskellBuildInputs = isHaskellPartition.right;
-      systemBuildInputs = isHaskellPartition.wrong;
+      inherit
+        buildDepends
+        buildTools
+        executableFrameworkDepends
+        executableHaskellDepends
+        executablePkgconfigDepends
+        executableSystemDepends
+        executableToolDepends
+        extraLibraries
+        libraryFrameworkDepends
+        libraryHaskellDepends
+        libraryPkgconfigDepends
+        librarySystemDepends
+        libraryToolDepends
+        pkgconfigDepends
+        setupHaskellDepends
+        ;
+    } // stdenv.lib.optionalAttrs doCheck {
```

Comment:
I don't think this needs to be optional given we are calculating regardless of whether doCheck / doBenchmark is on.